components/brave_wallet/browser/internal/polkadot/rust/sr25519.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_POLKADOT_RUST_SR25519_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_POLKADOT_RUST_SR25519_H_
+
+#include <array>
+#include <memory>
+
+#include "base/containers/span.h"
+
+namespace brave_wallet::polkadot {
+
+using SR25519PublicKey = std::array<uint8_t, 32>;
+using SR25519Signature = std::array<uint8_t, 64>;
+
+class SchnorrkelKeyPair {
```

Comment:
Does this class necessarily needs to be a virtual? If so, add a few comments explaining this pure virtual class, but if it is just the case of making sure you can mock tests, it would be better if we could test with fake data, or something similar, because it adds a certain cognitive load to follow virtualised code when trying to understand how things work.